js/lib.deno_runtime.d.ts

```diff
@@ -919,6 +919,9 @@ declare namespace Deno {
     network: Network;
     address: string;
   }
+  export interface NetworkOptions {
+    network: Network;
```

Comment:
I think this should be called "protocol"

Reply:
Or maybe this is just woefully over-engineered in the first place. What other networks are there? ATM😕?

Reply:
https://golang.org/pkg/net/#Dial

I'd prefer to stick with "network" since that's what Go uses

>  Known networks are "tcp", "tcp4" (IPv4-only), "tcp6" (IPv6-only), "udp", "udp4" (IPv4-only), "udp6" (IPv6-only), "ip", "ip4" (IPv4-only), "ip6" (IPv6-only), "unix", "unixgram" and "unixpacket".

Reply:
Thinking a bit more about it, let's consider some use cases:
  * use TLS or SSH to set up an encrypted connection (I would not call those options a 'network')
  * Unix domain sockets (decidedly not a network)
  * UDP (but that's connectionless, hence `dial()` is not applicable)
  * SCTP/IPSec (these are generally not availabe to user-mode applications)

Nothing that I can think of where 'network' is the obvious choice for a parameter name.

Reply:
> I'd prefer to stick with "network" since that's what Go uses
> 
> > Known networks are "tcp", "tcp4" (IPv4-only), "tcp6" (IPv6-only), "udp", "udp4" (IPv4-only), "udp6" (IPv6-only), "ip", "ip4" (IPv4-only), "ip6" (IPv6-only), "unix", "unixgram" and "unixpacket".

In the rest of the world this is known as "address family".

Reply:
My plan wasn't to fit TLS into this function but have a separate one. I think that's prudent because TLS has many more options than transport layer (level 4).

Reply:
> My plan wasn't to fit TLS into this function but have a separate one, dialTLS. I think that's prudent because TLS has many more options than transport layer (level 4).

That's true, I wasn't suggesting it'd be necessarily appropriate to support TLS in this context.
(On the other hand, it might be nice. Node.js differentiating between `http.get()` and `https.get()` was a painful mistake.)

Reply:
> On the other hand, it might be nice. Node.js differentiating between http.get() and https.get() was a painful mistake.

At least in that case, in deno those are both `fetch()`...